src/csg_test_harness/utilities.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+  *
+  *    Copyright (c) 2021 Project CHIP Authors
+  *
+  *    Licensed under the Apache License, Version 2.0 (the "License");
+  *    you may not use this file except in compliance with the License.
+  *    You may obtain a copy of the License at
+  *
+  *        http://www.apache.org/licenses/LICENSE-2.0
+  *
+  *    Unless required by applicable law or agreed to in writing, software
+  *    distributed under the License is distributed on an "AS IS" BASIS,
+  *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *    See the License for the specific language governing permissions and
+  *    limitations under the License.
+  */
+
+#include "constants.h"
+#include <string.h>
+
+// Message Payload (Initiator/Responder) keys
+std::string messageFromResponder_key ("MessageFromResponder");
+std::string messageFromInitiator_key ("MessageFromInitiator");
+
+// Request Parameter Keys
+std::string PBKDFParamRequest_key ("PBKDFParamRequest");
+
+// Response Parameter Keys
+std::string PBKDFParamResponse_key ("PBKDFParamResponse");
+std::string PBKDFParamResponse_salt_length_key ("SaltLength");
+std::string PBKDFParamResponse_iter_count_key ("IterCount");
+
+// PAKE Contribution (PAKE 1)
+std::string PAKE_1_key ("PAKE1");
+std::string PAKE_1_Pa_key ("PAKEPa");
+std::string PAKE_1_key_id_key ("PAKEKeyID");
+
+std::string PAKE_2_key ("PAKE2");
+std::string PAKE_2_encryption_id_key ("EncryptionID");
+
+std::string PAKE_3_key ("PAKE3");
+
+const char * yaml_string_for_map(std::map<std::string, std::map<std::string, std::string>> *trace_map)
```

Comment:
I wonder if there is a library or something that does this for us? Worried if there are any edge cases that might bite us.

Reply:
Filed https://github.com/chip-csg/connectedhomeip/issues/28 to investigate

Reply:
https://github.com/jbeder/yaml-cpp/blob/master/LICENSE